mailpoet/lib/Util/License/Features/CapabilitiesManager.php

```diff
@@ -0,0 +1,118 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Util\License\Features;
+
+use MailPoet\Config\ServicesChecker;
+use MailPoet\Settings\SettingsController;
+use MailPoet\Util\License\Features\Data\Capabilities;
+
+class CapabilitiesManager {
+  // Settings mapping
+  const MSS_TIER_SETTING_KEY = 'mta.mailpoet_api_key_state.data.tier';
+  const MSS_MAILPOET_LOGO_IN_EMAILS_SETTING_KEY = 'mta.mailpoet_api_key_state.data.mailpoet_logo_in_emails';
+  const MSS_DETAILED_ANALYTICS_SETTING_KEY = 'mta.mailpoet_api_key_state.data.detailed_analytics';
+  const MSS_AUTOMATION_STEPS_SETTING_KEY = 'mta.mailpoet_api_key_state.data.automation_steps';
+  const MSS_SEGMENT_FILTERS_SETTING_KEY = 'mta.mailpoet_api_key_state.data.segment_filters';
+  // Product capabilities mapping
+  const MIN_TIER_LOGO_NOT_REQUIRED = 1;
+  const MIN_TIER_ANALYTICS_ENABLED = 1;
+  const MIN_TIER_UNLIMITED_AUTOMATION_STEPS = 2;
+  const MIN_TIER_UNLIMITED_SEGMENT_FILTERS = 2;
+
+  private SettingsController $settings;
+  private ServicesChecker $servicesChecker;
+  private Subscribers $subscribersFeature;
+  private ?int $tier;
+  private bool $isKeyValid = false;
+
+  public function __construct(
+    SettingsController $settings,
+    ServicesChecker $servicesChecker,
+    Subscribers $subscribersFeature
+  ) {
+    $this->settings = $settings;
+    $this->servicesChecker = $servicesChecker;
+    $this->subscribersFeature = $subscribersFeature;
+  }
+
+  private function getTier(): ?int {
+    $tier = $this->settings->get(self::MSS_TIER_SETTING_KEY);
+    return isset($tier) ? (int)$tier : null;
+  }
+
+  private function isMailpoetLogoInEmailsRequired(): bool {
+    $mailpoetLogoInEmails = $this->settings->get(self::MSS_MAILPOET_LOGO_IN_EMAILS_SETTING_KEY);
+
+    if (!isset($this->tier) && !isset($mailpoetLogoInEmails)) {
+      return !$this->servicesChecker->isUserActivelyPaying(); // Backward compatibility
+    }
+
+    if (!$this->isKeyValid) {
+      return true;
+    }
+
+    // Allow for less restrictive individual capability to take precedence over tier
+    if (isset($mailpoetLogoInEmails) && (bool)$mailpoetLogoInEmails === false) {
+      return false;
+    }
+
+    return (isset($this->tier) && $this->tier < self::MIN_TIER_LOGO_NOT_REQUIRED) || (isset($mailpoetLogoInEmails) && (bool)$mailpoetLogoInEmails === true);
```

Comment:
As implemented here, a more restrictive individual capability also takes precedence over tier. For example, if `tier` is `1` (no logo) and `mailpoetLogoInEmails` is `true` (show logo), the logo will be shown. I'm not sure such a combination is designed to happen. But if we consider the implemented logic to be correct, it can be simplified to:

```suggestion
    // Allow for individual capability to take precedence over tier
    if (isset($mailpoetLogoInEmails)) {
      return (bool)$mailpoetLogoInEmails;
    }

    return (isset($this->tier) && $this->tier < self::MIN_TIER_LOGO_NOT_REQUIRED);
```

Reply:
> As implemented here, a more restrictive individual capability also takes precedence over tier. 

Thanks for the catch, that was wrong. Fixed.